### PR TITLE
Make logging unit test file a temp file

### LINKED ISSUE #2843: Logging unit tests create a file name with PID and then don't clean it up.
@waprin Can you fix?

For example

```
$ git st
On branch upgrade-grpcio
nothing to commit, working tree clean
$ tox -e py27
...
 $ git st
On branch upgrade-grpcio
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        14920

nothing added to commit but untracked files present (use "git add" to track)
$ cat 14920 | wc
      0       0       0
```

### PR DESCRIPTION
**Pull request by waprin**
Fixes #2843. This doesn't clean up the file after but it does make it a temp file, @dhermes let me know if you think this is sufficient which I think it is.

### CODE REVIEW

File: logging/unit_tests/test_client.py [hunk 1]
```diff
@@ -562,7 +563,8 @@ def test_get_default_handler_app_engine(self):
                                 credentials=_make_credentials(),
                                 use_gax=False)
 
-        with _Monkey(_MUT, _LOG_PATH_TEMPLATE='{pid}'):
+        temp_log_path = os.path.join(tempfile.mkdtemp(), '{pid}')
```
Comment (tseaver) [inline logging/unit_tests/test_client.py]: This leaks tempdirs every time the tests are run. :<

Comment (daspecster) [inline logging/unit_tests/test_client.py]: Well crap...
>The user of mkdtemp() is responsible for deleting the temporary directory and its contents when done with it.
- https://docs.python.org/2/library/tempfile.html#tempfile.mkdtemp

Fixing...

Comment (waprin) [inline logging/unit_tests/test_client.py]: Ya I mentioned that in PR comment, thought it wasn't a big deal cause it's in tmp but I supposed might as well clean it up, ty for fixing.

Comment (daspecster) [inline logging/unit_tests/test_client.py]: Yeah I saw that but I what I didn't realize was...
>The directory is readable, writable, and searchable only by the creating user ID.

@waprin thanks for pointing it out in your notes and thanks @tseaver for the fix!

**Pull request merged**